### PR TITLE
TST: Move 32-bit and parallel job to Actions

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to move "32-bit and parallel" job from CircleCI to GitHub Actions. I purposely let the full CI run to make sure I didn't mess up CircleCI config. (CircleCI seems to work as expected in the first commit, so now I will skip it.)

There were talks about we should stop using Docker image or something but I am not sure what to do with that part.

This is a follow up of #10388 and address part of #11038 .

### CODE REVIEW

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -108,3 +108,34 @@ jobs:
...
       run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
+
+  parallel_and_32bit:
+    name: 32-bit and parallel
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/pypa/manylinux1_i686
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (pllim) [inline .github/workflows/ci_workflows.yml:119]: Looks like this doesn't work with container?
```
/usr/bin/docker exec  .... sh -c "cat /etc/*release | grep ^ID"
OCI runtime exec failed: exec failed: container_linux.go:370: starting container process caused: no such file or directory: unknown
```

Comment (pllim) [reply .github/workflows/ci_workflows.yml]: Looks like a known and unresolved issue at actions/checkout#334

**Comment by pllim** [comment]
I think it works. But you should check carefully.

Actions logs: https://github.com/astropy/astropy/pull/11043/checks?check_run_id=1400152329
CircleCI logs: https://app.circleci.com/pipelines/github/astropy/astropy/3246/workflows/1b64c442-a496-4ba1-b769-34509c767e58/jobs/81931

**Comment by pllim** [comment]
Is there anything that is blocking this?

File: .github/workflows/ci_workflows.yml [hunk 2]
```diff
@@ -108,3 +108,35 @@ jobs:
...
+      with:
+        fetch-depth: 0
+    - name: Write configuration items to standard location to make sure they are ignored in parallel mode
+      run: |
+        mkdir -p $HOME/.astropy/config/
+        printf "unicode_output = True\nmax_width = 500" > $HOME/.astropy/config/astropy.cfg
+    # In addition to testing 32-bit, we also use the 3.7 builds to
+    # test the ability to run the test suite in parallel.
+    - name: Install dependencies for Python 3.7
+      run: /opt/python/cp37-cp37m/bin/pip install tox
```
Comment (astrojuanlu) [inline .github/workflows/ci_workflows.yml:130]: Question: Does

```suggestion
      run: python3.7 -m pip install tox
```

work instead? (Same below with Python 3.8) or are these paths not in `$PATH` by default?

Comment (astrojuanlu) [reply .github/workflows/ci_workflows.yml:130]: Oh, I see! It's a `quay.io/pypa/manylinux1_i686` thing :+1:

**Review by astrojuanlu — APPROVED** [review]
The Actions spec looks like a good translation of the Circle CI one. I went over the Actions logs and didn't spot anything suspicious.

File: .github/workflows/ci_workflows.yml (hunk 1 above)
Comment (astrojuanlu) [reply .github/workflows/ci_workflows.yml]: Subscribed!

**Review by saimn — APPROVED** [review]
LGTM.

**Comment by saimn** [comment]
I don't see a reason to delay this, so let's merge! Thanks @pllim.

**Comment by pllim** [comment]
Thanks for the reviews, y'all!

**Pull request merged**